src/energyplus/Test/GeometryTranslator_GTest.cpp

```diff
@@ -620,3 +626,470 @@ TEST_F(EnergyPlusFixture, GeometryTranslator_4ZoneWithShading_Simple_2) {
   ReverseTranslator reverseTranslator;
   EXPECT_NO_THROW(reverseTranslator.translateWorkspace(workspace));
 }
+
+
+
+
+Workspace makeShadingExampleWorkspace(CoordinateSystem coordinateSystem, double northAxis) {
+
+  // Test for #4111 - Bug importing Shading:Building and Shading:Site objects from IDF
+
+  // Test building, with 4 shading surfaces
+  //
+  //   y
+  //    ^
+  //    |
+  // 10 +  ┌───────────────────────────┐                              ┌───────────────────────────┐
+  //    |  │                           │                              │                           │
+  //    |  │   Shading:Site:Detailed   │                              │ Shading:Building:Detailed │
+  //    |  │                           │                              │                           │
+  //  5 +  └───────────────────────────╔══════════════════════════════╗───────────────────────────┘
+  //    |                              ║                              ║
+  //    |                              ║            Simple            ║
+  //    |                              ║           Building           ║
+  //    |                              ║                              ║
+  //  0 +  ┌───────────────────────────╚══════════════════════════════╝───────────────────────────┐
+  //    |  │                           │                              │                           │
+  //    |  │       Shading:Site        │                              │    Shading:Building       │
+  //    |  │                           │                              │                           │
+  // -5 +  └───────────────────────────┘                              └───────────────────────────┘
+  //    |
+  //    +--+---------------------------+------------------------------+---------------------------+-------> x
+  //      -10.                         0                              10                         20
```

Comment:
I'm doing four tests, each time testing the position of each of the 4 shading surfaces (which all have different types)

| CoordinateSystem | Building North Axis |
|------------------|---------------------|
| Relative         | 0                   |
| World            | 0                   |
| Relative         | 45                  |
| World            | 45                  |